core/lib/dal/src/connection/processor.rs

```diff
@@ -0,0 +1,222 @@
+use std::{
+    collections::HashMap,
+    fmt,
+    panic::Location,
+    sync::{
+        atomic::{AtomicUsize, Ordering},
+        Mutex,
+    },
+    time::{Duration, Instant, SystemTime},
+};
+
+use sqlx::{pool::PoolConnection, types::chrono, Connection, PgConnection, Postgres, Transaction};
+
+use crate::metrics::CONNECTION_METRICS;
+
+/// Tags that can be associated with a connection.
+#[derive(Debug, Clone, Copy)]
+pub(super) struct StorageProcessorTags {
+    pub requester: &'static str,
+    pub location: &'static Location<'static>,
```

Comment:
We could add user-defined tags (e.g., as `Vec<(&'static str, &dyn fmt::Debug)>`) as well similar to instrumenting queries, but I'm not sure whether it would bring much value.